src/main/java/com/minecolonies/apiimp/initializer/InteractionValidatorInitializer.java

```diff
@@ -52,8 +54,8 @@
     public static void init()
     {
         InteractionValidatorRegistry.registerStandardPredicate(new TranslationTextComponent(FURNACE_USER_NO_FUEL),
-          citizen -> citizen.getWorkBuilding() instanceof AbstractBuildingSmelterCrafter && ((AbstractBuildingSmelterCrafter) citizen.getWorkBuilding()).getCopyOfAllowedItems()
-                                                                                              .isEmpty());
+          citizen -> citizen.getWorkBuilding() instanceof AbstractBuildingSmelterCrafter
+                       && citizen.getWorkBuilding().getModule(GroupedItemListModule.class).get().getList(FUEL_LIST).isEmpty());
```

Comment:
You need todo a map on the optional here and then get the value via an OrElse().

This works completely against the Functional nature of Optional.